lib/gettext_i18n_rails/action_controller.rb

```diff
@@ -1,8 +1,23 @@
-class ActionController::Base
-  def set_gettext_locale
-    requested_locale = params[:locale] || session[:locale] || cookies[:locale] ||  request.env['HTTP_ACCEPT_LANGUAGE'] || I18n.default_locale
-    locale = FastGettext.set_locale(requested_locale)
-    session[:locale] = locale
-    I18n.locale = locale # some weird overwriting in action-controller makes this necessary ... see I18nProxy
+# Autoloading in initializers is deprecated on rails 6.0.  This delays initialization using the on_load
+# hooks, but does not change behaviour for existing rails versions.
+if defined?(Rails) && Rails::VERSION::MAJOR >= 6
+  ActiveSupport.on_load(:action_controller_base) do
+    class ::ActionController::Base
+      def set_gettext_locale
+        requested_locale = params[:locale] || session[:locale] || cookies[:locale] ||  request.env['HTTP_ACCEPT_LANGUAGE'] || I18n.default_locale
+        locale = FastGettext.set_locale(requested_locale)
+        session[:locale] = locale
+        I18n.locale = locale # some weird overwriting in action-controller makes this necessary ... see I18nProxy
+      end
+    end
   end
-end
+else
+  class ActionController::Base
+    def set_gettext_locale
+      requested_locale = params[:locale] || session[:locale] || cookies[:locale] ||  request.env['HTTP_ACCEPT_LANGUAGE'] || I18n.default_locale
+      locale = FastGettext.set_locale(requested_locale)
+      session[:locale] = locale
+      I18n.locale = locale # some weird overwriting in action-controller makes this necessary ... see I18nProxy
+    end
+  end
+end
```

Comment:
prefer doing a `path_controller = ->(){}` and then calling it from 2 places instead of copy-pasting all code

Reply:
Yep yep, makes sense.  I'll push a change for this momentarily.